akka-testkit-typed/src/main/scala/akka/testkit/typed/scaladsl/ActorTestKit.scala

```diff
@@ -0,0 +1,106 @@
+package akka.testkit.typed.scaladsl
+
+import akka.actor.typed.scaladsl.AskPattern._
+import akka.actor.typed.{ ActorRef, ActorSystem, Behavior, Props }
+import akka.annotation.{ ApiMayChange, InternalApi }
+import akka.testkit.typed.TestKitSettings
+import akka.testkit.typed.internal.{ ActorTestKitGuardian, TestKitUtils }
+import com.typesafe.config.{ Config, ConfigFactory }
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+object ActorTestKit {
+
+  def shutdown(
+    system:               ActorSystem[_],
+    duration:             Duration,
+    verifySystemShutdown: Boolean        = false): Unit =
+    TestKitUtils.shutdown(system, duration, verifySystemShutdown)
+
+}
+
+/**
+ * Testkit for asynchronous testing of typed actors, meant for mixing into the test class.
+ *
+ * Provides a typed actor system started on creation, used for all test cases and shut down when `shutdown` is called.
+ *
+ * The actor system has a custom guardian that allows for spawning arbitrary actors using the `spawn` methods.
+ *
+ * Designed to work with any test framework, but framework glue code that calls shutdown after all tests has
+ * run needs to be provided by the user.
+ *
+ * For synchronous testing of a `Behavior` see [[BehaviorTestKit]]
+ */
+@ApiMayChange
+trait ActorTestKit {
+  /**
+   * Actor system name based on the test it is mixed into, override to customize, or pass to constructor
+   * if using [[ActorTestKit]] rather than [[ActorTestKit]]
+   */
+  protected def name: String = TestKitUtils.testNameFromCallStack(classOf[ActorTestKit])
+
+  /**
+   * Configuration the actor system is created with, override to customize, or pass to constructor
+   * if using [[ActorTestKit]] rather than [[ActorTestKit]]
+   */
+  def config: Config = ConfigFactory.empty()
+
+  /**
+   * TestKit settings used in the tests, override or provide custom config to customize
+   */
+  protected implicit def testkitSettings = TestKitSettings(system)
+
+  private val internalSystem: ActorSystem[ActorTestKitGuardian.TestKitCommand] =
+    if (config == ConfigFactory.empty) ActorSystem(ActorTestKitGuardian.testKitGuardian, name)
+    else ActorSystem(ActorTestKitGuardian.testKitGuardian, name, config)
+
+  implicit final def system: ActorSystem[Nothing] = internalSystem
+
+  implicit def scheduler = system.scheduler
+  private val childName: Iterator[String] = Iterator.from(0).map(_.toString)
+
+  implicit val timeout = testkitSettings.DefaultTimeout
+
+  final def shutdownTestKit(): Unit = {
+    // FIXME separate default timeout for shutdown?
+    ActorTestKit.shutdown(system, timeout.duration)
```

Comment:
I think it can be important to fix this timeout, default 3 seconds might be too short

Reply:
I'll add another setting, for `TestKit.shutdown(system)` I hardcoded a value but I realize I could just create read that same setting in that case as well.